keycloak: 20.0.5 -> 21.1.1, keycloak-metrics-spi: 2.5.3 -> 3.0.0

ngerstle opened the pull request:
###### Description of changes

Based on https://github.com/NixOS/nixpkgs/pull/219010, with the added change to fix the metrics-spi plugin  to pass tests: https://github.com/NixOS/nixpkgs/pull/219010#issuecomment-1518587338

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

teutat3s reviewed (commented):
Thank you for this. Could you rebase to version 21.1.1 and add the changelogs to the commit, while you're at it? Nit: maybe split the two updates into two separate commits.

ngerstle committed:
keycloak-metrics-spi: 2.5.3 -> 3.0.0

Version increase: compatible with keycloak 21

changelog:
    feat: adds support for Keycloak 21 (thanks @ghenadiibatalski )
    fix: realm names are now shown again instead of ids (thanks @diversit )
    other fixes and improvements (thanks @claudiunicolaa , @douph1 , @uberspot)

ngerstle committed:
keycloak: 20.0.5 -> 21.1.1

Version increase: see
https://www.keycloak.org/docs/latest/release_notes/index.html#keycloak-21-0-0
for details and breaking changes.

ngerstle commented:
bundles https://github.com/NixOS/nixpkgs/pull/219010 as well

teutat3s commented on pkgs/servers/keycloak/keycloak-metrics-spi/default.nix:
```diff
@@ -2,11 +2,11 @@
 stdenv.mkDerivation rec {
   pname = "keycloak-metrics-spi";
-  version = "2.5.3";
+  version = "2.3.0";
```
While you fix this, please add the [changelog](https://github.com/aerogear/keycloak-metrics-spi/releases/tag/3.0.0) to the commit message as well.

teutat3s replied:
With this change, `nix-build nixos/tests/keycloak.nix` looks green here.

teutat3s commented:
Result of `nixpkgs-review pr 229344` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 test built:</summary>
  <ul>
    <li>nixosTests.keycloak.postgres</li>
  </ul>
</details>

talyz commented:
Tests pass, can't find any issues when deploying to my test environment. Thanks!
